kura/org.eclipse.kura.cloudconnection.sparkplug.mqtt.provider/src/main/java/org/eclipse/kura/cloudconnection/sparkplug/mqtt/message/SparkplugBProtobufPayloadBuilder.java

```diff
@@ -0,0 +1,175 @@
+        case Int16:
+            metricBuilder.setIntValue((Short) value);
+            break;
+        case Int32:
+            metricBuilder.setIntValue((Integer) value);
+            break;
+        case Int64:
+            metricBuilder.setLongValue((Long) value);
+            break;
+        case String:
+        case Text:
+        case UUID:
+            metricBuilder.setStringValue((String) value);
+            break;
+        case DateTime:
+            metricBuilder.setLongValue(((Date) value).getTime());
+            break;
+        case UInt8:
+            metricBuilder.setIntValue(Short.toUnsignedInt((Short) value));
+            break;
+        case UInt16:
+            metricBuilder.setIntValue((int) Integer.toUnsignedLong((Integer) value));
+            break;
+        case UInt32:
+            metricBuilder.setLongValue(Long.parseUnsignedLong(Long.toUnsignedString((Long) value)));
+            break;
+        case UInt64:
+            metricBuilder.setLongValue(((BigInteger) value).longValue());
+            break;
+        case DataSet:
+        case Template:
+        case PropertySet:
+        case PropertySetList:
+        case File:
+        case BooleanArray:
+        case DateTimeArray:
+        case UInt8Array:
+        case UInt64Array:
+        case UInt32Array:
+        case UInt16Array:
+        case StringArray:
+        case Int8Array:
+        case Int64Array:
+        case Int32Array:
+        case Int16Array:
+        case FloatArray:
+        case DoubleArray:
+        case Unknown:
+        default:
+            throw new UnsupportedOperationException("DataType " + dataType.toString() + " not implemented");
+        }
+
+        this.payloadBuilder.addMetrics(metricBuilder.build());
+
+        return this;
+    }
+
+    public <T> SparkplugBProtobufPayloadBuilder withMetric(String name, TypedValue<T> value, long timestamp) {
+        DataType sparkplugDataType;
+
+        switch (value.getType()) {
+        case BOOLEAN:
+            sparkplugDataType = DataType.Boolean;
+            break;
+        case BYTE_ARRAY:
+            sparkplugDataType = DataType.Bytes;
+            break;
+        case DOUBLE:
+            sparkplugDataType = DataType.Double;
+            break;
+        case FLOAT:
+            sparkplugDataType = DataType.Float;
+            break;
+        case INTEGER:
+            sparkplugDataType = DataType.Int32;
+            break;
+        case LONG:
+            sparkplugDataType = DataType.Int64;
+            break;
+        case STRING:
+            sparkplugDataType = DataType.String;
+            break;
+        default:
+            sparkplugDataType = DataType.Unknown;
+            break;
+        }
+
+        return this.withMetric(name, value.getValue(), sparkplugDataType, timestamp);
+    }
+
+    public <T> SparkplugBProtobufPayloadBuilder withMetric(String name, TypedValue<T> value) {
+        return this.withMetric(name, value, new Date().getTime());
+    }
```

Comment:
Is it correct? If it is only for time value, maybe it is better to change the name of the method.

Reply:
Its a lefover, will remove